virtualmachine/vsphere/vm.go

```diff
@@ -885,7 +885,14 @@ func (vm *VM) Destroy() (err error) {
 					err = e
 					break
 				}
-				if state == "notRunning" {
+
+				powerState, e := getPowerState(vm)
```

Comment:
we don't have to check powerstate unless gueststate is 'notRunning'. 
Once guestState is 'notRunning', break if powerstate is poweredOff else continue.